### DISCUSSION TITLE
System for color themes?

### DISCUSSION THREAD

**Discussion by nemanjam** [tailwindlabs/tailwindcss#7294] [Help]
Do you know where I could find a tutorial or example of how to define colors for multiple color themes, e.g. default-light, default-dark, light-blue, dark-blue, light-green, etc.? So light, dark, but also color themes. I work with Tailwind of course.

Specifically, what interests me are descriptive classes for colors, e.g. primary, secondary, accent, background, background-secondary, text, text-muted, etc., it is not entirely clear to me how many of these variants there are, whether some overlap between themes, how and whether to automatically create inverted colors, muted, light , dark variants and all other possible details I haven't mentioned. I think I have a basic idea, but I would not like to reinvent the wheel, but to adopt some already existing best practice, standard, an established system.

Note: I don't need a color palette - shades, I work with the default Tailwind color palette. I also don't need instructions on how to switch themes, I already have that implemented.

If you know a useful guide or example, please post a link. Maybe I am not using right words on Google.

**Comment by walidvb** [comment]
I posted a more generic question around theming that might be relevant to you, with a proof of concept: https://github.com/tailwindlabs/tailwindcss/discussions/7536
